src/app/download-list/download-list.page.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { NavController } from '@ionic/angular';
+import { AlertService } from '../services/alert/alert.service';
+import { DbService } from '../services/db/db.service';
+
+@Component({
+  selector: 'app-download-list',
+  templateUrl: './download-list.page.html',
+  styleUrls: ['./download-list.page.scss'],
+})
+export class DownloadListPage implements OnInit {
+  constructor(private navCtrl: NavController,private router:Router,private alertService:AlertService,private dbServices:DbService ) { }
+  resultsWithDownload: any[]=[];
+
+  async ngOnInit() {
+    await this.dbServices.openDatabase();
+    this.getdata();
+  }
+
+  async getdata(){
+    this.resultsWithDownload = await this.dbServices.getAllTransactions()
+  }
+
+  goBack() {
+    this.navCtrl.back();
+  }
+
+  navigate(data:any) {
+    this.router.navigate(['project-details'], { state: { _id:data || null} });
```

Comment:
What are we passing null here?
"id" will be there right everytime

Reply:
null is the type of the page we are switching.
id will be there everyTime, yes